2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala

```diff
@@ -127,6 +127,7 @@ package model {
   final case class CollectorConfig(
     interface: String,
     port: Int,
+    paths: Map[String, String] = Map.empty[String, String],
```

Comment:
Default parameters generally considered a bad practice in Scala https://blog.ssanj.net/posts/2019-05-01-why-are-default-parameter-values-considered-bad-in-scala.html. Especially if they're not in the end.

Reply:
Changed it to `Option[Map[String, String]]`.

Reply:
But `None` is not different from empty `Map`?

Reply:
Sure, but then:

```
val mappings = config.paths.getOrElse(Map.empty[String, String])
```

Reply:
Exactly. And we're ending up with empty map by default anyways, unless I'm missing something very obvious.

Reply:
Maybe I am...

Your note was about default parameters, especially when not last in the list. We are no longer using them. The `paths` parameter is an optional map. You cannot look for a value by key in a `None`, so if the map is not provided, we use an empty map instead.

Do you still have any concerns?